horovod/tensorflow/__init__.py

```diff
@@ -243,23 +242,27 @@ def __init__(self, optimizer, name=None, use_locking=False, device_dense='',
             self._allreduce_grads = _make_allreduce_grads_fn(
                 name, device_dense, device_sparse, compression, sparse_as_dense)
 
-        def apply_gradients(self, grads_and_vars, *args, **kwargs):
-            """Apply gradients to provided variables.
+        def compute_gradients(self, *args, **kwargs):
+            """Compute gradients of all trainable variables.
 
-            See Optimizer.apply_gradients() for more info.
+            See Optimizer.compute_gradients() for more info.
 
-            In DistributedOptimizer, apply_gradients() is overriden to also
-            allreduce the gradients before applying them.
+            In DistributedOptimizer, compute_gradients() is overriden to also
+            allreduce the gradients before returning them.
             """
+            self._fail_if_executing_eagerly()
```

Comment:
Same here. Is it possible to just have the eager failure occur during `__init__` instead of in both `apply_gradients` and `compute_gradients`?